### PR TITLE
Add tests for ACCEPT *

### PR DESCRIPTION
**Pull request by 82marbag**
*Issue #, if available:* https://github.com/awslabs/smithy/issues/1332

*Description of changes:* Add protocol tests for the new ACCEPT header behavior. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AndrewFossAWS — APPROVED** [review]

File: smithy-aws-protocol-tests/model/restJson1/http-accept.smithy [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        id: "AcceptHeaderStarRequestTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/test-accept-header",
+        headers: {
+            "Accept": "application/*",
+        },
+        params: {},
+        body: "{}",
+        method: "GET",
+		appliesTo: "server",
```
Comment (kstich) [inline smithy-aws-protocol-tests/model/restJson1/http-accept.smithy:23]: This use tabs and not spaces.

File: smithy-aws-protocol-tests/model/restJson1/http-accept.smithy [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        id: "AcceptHeaderStarStarRequestTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/test-accept-header",
+        headers: {
+            "Accept": "*/*",
+        },
+        params: {},
+        body: "{}",
+        method: "GET",
+		appliesTo: "server",
```
Comment (kstich) [inline smithy-aws-protocol-tests/model/restJson1/http-accept.smithy:35]: This use tabs and not spaces.

**Comment by 82marbag** [comment]
@kstich updated with spaces

File: smithy-aws-protocol-tests/model/restJson1/http-accept.smithy [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+@httpRequestTests([
+    {
+        id: "AcceptHeaderStarRequestTest",
+        protocol: "aws.protocols#restJson1",
+        uri: "/test-accept-header",
+        headers: {
+            "Accept": "application/*",
+        },
+        params: {},
+        body: "{}",
```
[the remark this answers is not in the document]
Comment (82marbag) [reply smithy-aws-protocol-tests/model/restJson1/http-accept.smithy]: Updated

**Review by kstich — APPROVED** [review]

**Pull request merged**